app/models/billing_profile.rb

```diff
@@ -1,19 +1,16 @@
 require 'countries'
 
 class BillingProfile < ApplicationRecord
+  include BillingHelpers
   alias_attribute :country_code, :alpha_two_country_code
 
   validates :name, presence: true
   validates :country_code, presence: true
   validates :vat_code, uniqueness: { scope: :user_id }, allow_blank: true
 
   belongs_to :user, optional: true
-
-  def address
-    country_name = Countries.name_from_alpha2_code(country_code)
-    postal_code_with_city = [postal_code, city].join(' ')
-    [street, postal_code_with_city, state, country_name].compact.join(', ')
-  end
+  before_destroy :check_no_active_offers_associated
```

Comment:
Any chance to avoid callbacks? It comes difficult to maintain such code. Perhaps introducing `deletable?` predicate can help?